agent/api/task.go

```diff
@@ -97,6 +97,15 @@ type Task struct {
 	KnownStatusTimeUnsafe time.Time `json:"KnownTime"`
 	knownStatusTimeLock   sync.RWMutex
 
+	// PullStartedAt is the timestamp when the task start pulling
```

Comment:
This needs a longer description. Especially because there's no `pull` for the task (same with other timestamp variables you've added to this struct.